internal/providers/vault/README.md

```diff
@@ -0,0 +1,88 @@
+# Vault Provider
+
+The Vault provider for Secretless can fetch secrets from configured backends 
+in [HashiCorp Vault](https://www.vaultproject.io). The provider is based on 
+the [Vault API client](https://pkg.go.dev/github.com/hashicorp/vault/api) in Go. 
+It reads values from the configured paths.
+
+## Usage Documentation
+
+The Vault provider is configured in the `secretless.yml`. Using
+`from: vault` and `get: /path/to/secret/in/vault` for the credentials in the 
+service connector configuration. Note that Vault supports various backends 
+that store different kinds of secrets. There is the cubbyhole, KV v1 and v2, 
+database, etc.
```

Comment:
Here we should mention which ones we support.

Reply:
Oh I'm not suggesting adding additional support for the others, just that we mention which ones are supporting (or have confidence that they work)